webapp/channels/src/components/widgets/modals/components/checkbox_setting_item.tsx

```diff
@@ -1,35 +1,40 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import type {ReactNode} from 'react';
 import React from 'react';
-import {FormattedMessage} from 'react-intl';
-import type {MessageDescriptor} from 'react-intl';
 
 import type {BaseSettingItemProps} from './base_setting_item';
 import BaseSettingItem from './base_setting_item';
 
 export type FieldsetCheckbox = {
     dataTestId?: string;
-    title: MessageDescriptor;
     name: string;
 }
 
 type Props = BaseSettingItemProps & {
     inputFieldData: FieldsetCheckbox;
     inputFieldValue: boolean;
+
+    /**
+     * The title of the checkbox input field, pass in FormattedMessage component for styling compatibility
```

Comment:
Is there a reason we're wanting people to pass in `FormattedMessage` instead of just providing a `id` and `description` (or just call `formatMessage`) so this component can generate it itself?

Reply:
Apologies i should have mentioned summary in the PR description. I have added it now.